spec/controllers/blocks_controller_spec.rb

```diff
@@ -0,0 +1,198 @@
+require 'rails_helper'
+
+RSpec.describe BlocksController, type: :controller do
+  describe "GET index" do
+    it "requires login" do
+      get :index
+      expect(response).to redirect_to(root_url)
+    end
+
+    it "succeeds" do
+      login
+      get :index
+      expect(response.status).to eq(200)
+    end
```

Comment:
Can we test this succeeds _also_ with extant blocks?

Ideally we'd feature spec this, too, so we load the views up.